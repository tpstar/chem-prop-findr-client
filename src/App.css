
/*  SECTIONS  ============================================================================= */

.section {
	clear: both;
	padding: 0px;
	margin: 0px;
  font-size: 20px;
}

/*  GROUPING  ============================================================================= */

.row {
    zoom: 1; /* For IE 6/7 (trigger hasLayout) */
}

.row:before,
.row:after {
    content:"";
    display:table;
}

.row:after {
    clear:both;
}



/*  GRID COLUMN SETUP   ==================================================================== */

.col {
	display: block;
	float:left;
	margin: 1% 0 1% 1.6%;
}

.col:first-child { margin-left: 0; } /* all browsers except IE6 and lower */


/*  REMOVE MARGINS AS ALL GO FULL WIDTH AT 480 PIXELS */

@media only screen and (max-width: 480px) {
	.col {
		/*margin: 1% 0 1% 0%;*/
        margin: 0;
	}
}


/*  GRID OF TWO   ============================================================================= */


.span-2-of-2 {
	width: 100%;
}

.span-1-of-2 {
	width: 49.2%;
}

/*  GO FULL WIDTH AT LESS THAN 480 PIXELS */

@media only screen and (max-width: 480px) {
	.span-2-of-2 {
		width: 100%;
	}
	.span-1-of-2 {
		width: 100%;
	}
}

/*  GRID OF THREE   ============================================================================= */


.span-3-of-3 {
	width: 100%;
}

.span-2-of-3 {
	width: 66.13%;
}

.span-1-of-3 {
	width: 32.26%;
}


/*  GO FULL WIDTH AT LESS THAN 480 PIXELS */

@media only screen and (max-width: 480px) {
	.span-3-of-3 {
		width: 100%;
	}
	.span-2-of-3 {
		width: 100%;
	}
	.span-1-of-3 {
		width: 100%;
	}
}

#root {
  background-image: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7)), /* make background hazy */
                    url(img/tpstarzrbn3.png);
  background-size: cover;
  background-position: center;
  height: 100vh; /* fill whole page with header */
  background-attachment: fixed;
}

.row {
  max-width: 1140px;
  margin: 0 auto;
}

.logo {
    height: 80px;
    width: auto;
    float: left;
    margin-top: 20px;
}

.search-chemical {
  padding-top: 20px;
}



.chemical {
  padding-top: 20px;
}

.conversion {
  padding-top: 10px;
}

.conversion:before {
    display: block;
    height: 2px;
    background-color: #6F79A8;
    content: " ";
    width: 150px;
    margin: 5px auto 20px;
}

h1,
h3 {
  color: #9ea7d8;
}

h3 {
  margin-bottom: 20px;
}

p {
  color: #D0D8ff;
  /* font-size: 120%; */
}

.main-text {
  margin-top: 35%;
}

.wrong-name {
  color: #FFE082;
}


.panel {
  border-color: CornFlowerBlue;
  font-size: 115%;
}
