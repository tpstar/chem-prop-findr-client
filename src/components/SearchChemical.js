import React, { Component } from 'react';
import { connect } from "react-redux";
import { fetchChemical } from "../actions/index";
import { FormControl } from 'react-bootstrap';

export class SearchChemical extends Component {
  constructor(props) {
    super(props)
    this.state = { term: ""};
    this.handleOnChange = this.handleOnChange.bind(this)
    this.handleOnSubmit = this.handleOnSubmit.bind(this)
  }

  handleOnChange(event) {
    this.setState({term: event.target.value})
  }

  handleOnSubmit(event) {
    console.log(this.state.term);
    event.preventDefault(); // prevent the page from refreshing
    this.props.fetchChemical(this.state.term);
    this.setState({term: ""})
                  // initialize the search term
  }

  render() {
    return (
      <div className="search-chemical">
        <h3>Search</h3>
        <form onSubmit={this.handleOnSubmit}>
          <FormControl
            bsSize="lg"
            value={this.state.term}
            type="text"
            className="form-control"
            placeholder="Enter the name of a chemical"
            onChange={this.handleOnChange}
            // onKeyPress={this.handleKeyPress}
          />
        </form>
      </div>
    )
  }
}

export default connect(null,{ fetchChemical })(SearchChemical);
