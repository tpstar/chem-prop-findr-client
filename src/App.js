import React, { Component } from 'react';
import Chemical from './components/Chemical'
import SearchChemical from './components/SearchChemical';
import Conversion from './components/Conversion'
import ChemicalService from './services/ChemicalService';
import './App.css';

class App extends Component {
  constructor() {
    super()

    this.state = {
      chemical: {name: " "},
      isNew: true
    }

    //this.searchChemical = this.searchChemical.bind(this)
  }

  //searchChemical(chemical) {
  searchChemical = chemical => {
    ChemicalService.search(chemical)
      .then(chemical => {
        this.setState({chemical,
                       isNew: true})
      });

  }

  render() {
    return (
      <div className="App section">
        <div className='row'>
          <img src={require('./img/hanlee_logo.png')} alt="hanlee logo" className="logo"/>
        </div>
        <div className='row'>
          <div className='col span-1-of-2'>
            <div className='main-text'>
              <h1>Find Chemical Properties of </h1>
              <h1>Chemicals</h1>
            </div>
          </div>
          <div className='col span-1-of-2'>
            <SearchChemical qChemical={this.searchChemical}/>

            <Chemical chemical={this.state.chemical}/>

            <Conversion chemical={this.state.chemical} isNew={this.state.isNew}/>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
