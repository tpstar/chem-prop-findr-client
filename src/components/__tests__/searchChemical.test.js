import React from 'react';
import SearchChemical from '../SearchChemical';
import renderer from 'react-test-renderer';
import '../../App.css';

describe('SearchChemical', () => {
	it('renders loading correctly', () => {
		const tree = renderer.create(
			<SearchChemical qChemical={""}/>
		).toJSON();
    console.log(tree);
		expect(tree).toMatchSnapshot();
	});
});
