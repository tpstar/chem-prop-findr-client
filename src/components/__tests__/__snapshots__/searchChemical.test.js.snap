// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchChemical renders loading correctly 1`] = `
<div
  className="search-chemical"
>
  <h3>
    Search
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      className="form-control form-control input-lg"
      onChange={[Function]}
      placeholder="Enter the name of a chemical"
      type="text"
      value=""
    />
  </form>
</div>
`;
